test/integration/builder/additional.js

```diff
@@ -112,6 +112,61 @@ module.exports = function(knex) {
 
     });
 
+    // TODO: This doesn't work on oracle yet.
+    if (['postgresql', 'mssql'].includes(knex.client.dialect)) {
+      describe('returning with wrapIdentifier and postProcessResponse`', () => {
+        let origHooks = {};
+
+        before('setup custom hooks', () => {
+          origHooks.postProcessResponse = knex.client.config.postProcessResponse;
+          origHooks.wrapIdentifier = knex.client.config.wrapIdentifier;
+
+          // Add `_foo` to each identifier.
+          knex.client.config.postProcessResponse = (res) => {
+            if (Array.isArray(res)) {
+              return res.map(it => {
+                if (typeof it === 'object') {
+                  return _.mapKeys(it, (value, key) => {
+                    return key + '_foo';
+                  });
+                } else {
+                  return it;
+                }
+              })
+            } else {
+              return res;
+            }
+          };
+
+          // Remove `_foo` from the end of each identifier.
```

Comment:
Symmetric transformation which removes and then re-creates suffix is not ideal to do here, since someone might just break calling wrap / postprocess altogether and test wouldn't know difference or someone may accidentally remove place where handlers are overwritten and still test runs fine. 

Also I had to went to look into handler code after checking the test code and wondering why keys weren't transformed to anything :)

Reply:
Shouldn't there be another test that tests whether the functions are called at all? This test only test if the functions are called correctly provided that the feature works.